### DISCUSSION TITLE
Consider different UX for OPDS feeds

### DISCUSSION THREAD

**Discussion by stevenzeck** [readium/kotlin-toolkit#48] [General]
In working on #403 I found that doing the layouts for OPDS feeds was a pain.  Right now it's all of this on a single screen:

1. 0 or more buttons for the feed navigation
2. A list of images in a grid layout for the feed publications
3. 0 or more lists of images in a horizontal linear layout list. Each one has a title and a "More" text/button. This is for the publications in a group
4. 0 or more buttons for the group navigation
5. Facets are placed in the toolbar menu as "FILTER"

I don't have a good design for this right now.  But I think this whole experience can be improved.  Maybe look to other implementations for inspiration?

**Comment by mickael-menu** [comment]
What do you think should be improved exactly? Is it the fact that we have different components on the same screen? This matches the content of mixed OPDS 2 feed.

I can't find it in the spec, but I always thought that we could have either `groups` OR a single pair (`navigation` and/or `publications`), which should simplify things a bit (@HadrienGardeur do you confirm?). In this case, here are the various layouts we can have for a feed:

* **Navigation** A list of sub-feeds
* **Publications** A vertically scrolled list of publications, with optionally a navigation list of sub-feeds at the top ([maybe as chips](https://material.io/components/chips)?)
* **Mixed** A vertically scrolled list with two kinds of cells:
    * **NavigationCell** A list of sub-feeds, maybe as chips to make it nicer?
    * **PublicationsCell** A horizontally scrolled carousel with a title and a "More" button to expand it to a **Publications** layout

You can register to the beta test of Aldiko to check how we implemented this: https://play.google.com/apps/testing/com.aldiko.android

**Reply by stevenzeck** [reply]
It may just be the limited number of OPDS feeds I've worked with thus far, but I find it hard to navigate through them.  If the feed publication list has a lot of items, I don't even know if there are group publications or navigations for a group.  The navigation buttons definitely take up too much space.

I'm thinking of better ways to present them, but I feel what the testapp does right now can be vastly improved.

**Reply by mickael-menu** [reply]
Yes, I agree that the way navigation links are presented is not very appealing.
